File: sdk/adapters/inmobi/InMobiAdapter.kt
package com.rivalapexmediation.adapter.inmobi

import android.content.Context
import com.rivalapexmediation.sdk.adapter.AdNetworkAdapter
import com.rivalapexmediation.sdk.adapter.AdLoadCallback
import com.rivalapexmediation.sdk.models.Ad
import com.rivalapexmediation.sdk.models.AdType
import java.util.concurrent.atomic.AtomicBoolean

/**
 * InMobi Adapter for Rival ApexMediation
 */
class InMobiAdapter : AdNetworkAdapter {
    override val networkName = "inmobi"
    override val version = "1.0.0"
    override val minSDKVersion = "1.0.0"
    
    private val isInitialized = AtomicBoolean(false)
    
    override fun initialize(context: Context, config: Map<String, Any>) {
        if (isInitialized.get()) return
        // TODO: Initialize InMobi SDK
        // InMobiSdk.init(context, accountId, consentObject)
        isInitialized.set(true)
    }
}

File: sdk/adapters/inmobi/InMobiAdapter.swift
import Foundation

@objc(RAMInMobiAdapter)
public class InMobiAdapter: NSObject, RAMAdNetworkAdapter {
    public let networkName = "inmobi"
    public let version = "1.0.0"
    public let minSDKVersion = "1.0.0"
    
    private var isInitialized = false
    
    public func initialize(with config: [String: Any], completion: @escaping (Error?) -> Void) {
        if isInitialized {
            completion(nil)
            return
        }
        
        // TODO: Initialize InMobi SDK
        // IMSdk.initWithAccountID(accountId, consentDictionary: consent)
        isInitialized = true
        completion(nil)
    }
}
